addons/account/models/account_move.py

```diff
@@ -502,12 +502,41 @@ def _onchange_partner_id(self):
 
         # Find the new fiscal position.
         delivery_partner_id = self._get_invoice_delivery_partner_id()
+        # Keep track of whether the fiscal position has changed
+        previous_fiscal_position_id = self.fiscal_position_id
         self.fiscal_position_id = self.env['account.fiscal.position'].get_fiscal_position(
             self.partner_id.id, delivery_id=delivery_partner_id)
-        self._recompute_dynamic_lines()
+
+        # Update the taxes if the fiscal position has changed
+        if previous_fiscal_position_id != self.fiscal_position_id:
+            self._onchange_fiscal_position_id()
+        else:
+            self._recompute_dynamic_lines()
+
         if warning:
             return {'warning': warning}
 
+    @api.onchange('fiscal_position_id')
+    def _onchange_fiscal_position_id(self):
+        for line in self.line_ids:
+            # Only map map fiscal positions from lines with products that have defined taxes
+            if (
+                not line.product_id.taxes_id
+                and self.move_type in ('out_invoice', 'out_refund')
+                or (
+                    not line.product_id.supplier_taxes_id
+                    and self.move_type in ('in_invoice', 'in_refund')
+                )
+                or line.display_type in ('line_section', 'line_note')
+            ):
+                continue
+            line.tax_ids = line._get_computed_taxes()
+            line.account_id = line._get_computed_account()
```

Comment:
But `_get_computed_taxes` is not taking care of the fiscal position?

You can see that every single time when it is called, something similar is done afterwards
```python
            taxes = line._get_computed_taxes()
            if taxes and line.move_id.fiscal_position_id:
                taxes = line.move_id.fiscal_position_id.map_tax(taxes)
```

That should probably be done inside of `_get_computed_taxes` @smetl ?